docs/readmes/bazel/bazel.md

```diff
@@ -0,0 +1,510 @@
+- `cc_test` and `go_test` are used for C, C++ and Go unit tests.
+- `pytest_test` is a wrapper around `py_test`. This rule type is used for Python unit tests, the [LTE integration tests](#lte-integration-tests) and the [Python sudo tests](#python-sudo-tests).
+- `proto_library`, `cpp_proto_library`, `python_proto_library`, `cpp_grpc_library`, `python_grpc_library` are used to group proto code, as well as for providing proto and grpc code for C++ and Python respectively.
+
+#### Target visibility
+
+Bazel targets have a [visibility attribute](https://bazel.build/concepts/visibility). Per default a target's visibility is set to private and the target is only accessible for targets within the same package. Possible values include (among others) a public visibility `//visibility:public` (accessible from all packages) or package specific visibility `//foo/bar:__pkg__`, to grant targets in package `//foo/bar` access. The visibility can be set explicitly for a target or globally for the whole package.
+
+The following coding guidelines should be applied when adding new targets. One important motivation for the guidelines below is based on encapsulation, i.e. keeping a clean architecture by restricting access to targets. For example, service libraries should only be accessed by the service itself. Proto and utility libraries can be accessed by any service (with certain restrictions if appropriate - e.g. the `all_python_proto` target should only be accessed by the `state` service).
+
+- Protos `BUILD.bazel` files: The global package visibility is set to public as compiled protos libraries need to be accessed from a lot of packages. The visibility of native proto libraries is set to private or is restricted to package visibility for other protos packages.
+- Service `BUILD.bazel` files: The global package visibility is set to private (i.e. default). The binaries should explicitly be set to private for readability reasons, even if this means redundancy. Libraries should explicitly be set to private or package visibility (e.g. for testing) when needed. The public visibility should be used only if necessary.
+- Utility Packages: In the `orc8r/configuration` and `orc8r/common` `BUILD.bazel` files the global package visibility is set to public.
+
+### Target command line syntax
+
+A complete description of the [syntax for specifying targets](https://bazel.build/run/build) can be found in the official bazel documentation. Here we provide a list of some useful examples:
+
+- To build all targets in the bazel workspace, excluding the ones tagged as manual, use `bazel build ...` or `bazel build //...`.
+- To build all targets inside of a folder, e.g. `lte/gateway`, use `bazel build //lte/gateway/...`.
+- Build one target `bazel build //lte/gateway/python/magma/mobilityd:mobilityd`, here the `//` indicate the start of the path to the target and the `:` indicated the start of the target name.
```

Comment:
Nit: you use `indicate` and then `indicated`, probably just stick to one (personally I would prefer `indicates`, but that is maybe just preference).